File: src/main/java/com/gymmanagement/service/MemberService.java
package com.gymmanagement.service;

import com.gymmanagement.dto.MemberRequestDTO;
import com.gymmanagement.dto.MemberResponseDTO;

import java.util.List;

public interface MemberService {

    MemberResponseDTO addMember(MemberRequestDTO memberRequestDTO);

    List<MemberResponseDTO> getAllMembers();

    MemberResponseDTO getMemberById(Long id);

    MemberResponseDTO updateMember(Long id, MemberRequestDTO memberRequestDTO);

    void deleteMember(Long id);

    List<MemberResponseDTO> searchMembers(String keyword);

    List<MemberResponseDTO> getMembersWithPagination(
            int page,
            int size,
            String sortBy,
            String direction
    );
}

File: frontend/src/services/memberService.js
import api from "./api";

const getAllMembers = async () => {
  const response = await api.get("/members");
  return response.data;
};

const getMemberById = async (id) => {
  const response = await api.get(`/members/${id}`);
  return response.data;
};

const addMember = async (memberData) => {
  const response = await api.post("/members", memberData);
  return response.data;
};

const updateMember = async (id, memberData) => {
  const response = await api.put(`/members/${id}`, memberData);
  return response.data;
};

const deleteMember = async (id) => {
  const response = await api.delete(`/members/${id}`);
  return response.data;
};

const searchMembers = async (keyword) => {
  const response = await api.get("/members/search", {
    params: {
      keyword,
    },
  });

  return response.data;
};

const getMembersWithPagination = async (
  page = 0,
  size = 5,
  sortBy = "memberId",
  direction = "asc"
) => {
  const response = await api.get("/members/page", {
    params: {
      page,
      size,
      sortBy,
      direction,
    },
  });

  return response.data;
};

const memberService = {
  getAllMembers,
  getMemberById,
  addMember,
  updateMember,
  deleteMember,
  searchMembers,
  getMembersWithPagination,
};

export default memberService;